codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithyrust/generator/RustAwsSdkShimGenerator.java

```diff
@@ -0,0 +1,1739 @@
+      dafnyTypesModuleName,
+      "fluentMemberSetters",
+      fluentMemberSettersForStructure(inputShape).toString()
+    );
+
+    return TokenTree.of(
+      evalTemplate(
+        """
+        #[allow(dead_code)]
+        pub fn from_dafny(
+            dafny_value: ::std::rc::Rc<
+                crate::r#$dafnyTypesModuleName:L::$structureName:L,
+            >,
+            client: $sdkCrate:L::Client,
+        ) -> $sdkCrate:L::operation::$snakeCaseOperationName:L::builders::$operationName:LFluentBuilder {
+            client.$snakeCaseOperationName:L()
+                  $fluentMemberSetters:L
+        }
+        """,
+        variables
+      )
+    );
+  }
+
+  private TokenTree fluentMemberSettersForStructure(Shape shape) {
+    return TokenTree
+      .of(
+        shape
+          .members()
+          .stream()
+          .map(m ->
+            TokenTree.of(
+              ".set_" +
+              toSnakeCase(m.getMemberName()) +
+              "(" +
+              fromDafnyForMember(m) +
+              ")"
+            )
+          )
+      )
+      .lineSeparated();
+  }
+
+  private RustFile operationResponseConversionModule(
+    final OperationShape operationShape
+  ) {
+    var operationModuleName = toSnakeCase(operationShape.getId().getName());
+
+    var toDafnyFunction = operationResponseToDafnyFunction(operationShape);
+
+    return new RustFile(
+      Path.of(
+        "src",
+        "conversions",
+        operationModuleName,
+        "_" + operationModuleName + "_response.rs"
+      ),
+      toDafnyFunction
+    );
+  }
+
+  private TokenTree operationResponseToDafnyFunction(
+    final OperationShape operationShape
+  ) {
+    StructureShape outputShape = model.expectShape(
+      operationShape.getOutputShape(),
+      StructureShape.class
+    );
+    String operationName = operationShape.getId().getName();
+    String snakeCaseOperationName = toSnakeCase(operationName);
+    String sdkId = service
+      .expectTrait(ServiceTrait.class)
+      .getSdkId()
+      .toLowerCase();
+    String dafnyTypesModuleName =
+      "software::amazon::cryptography::services::%s::internaldafny::types".formatted(
+          sdkId
+        );
+    String structureName = outputShape.getId().getName();
+    Map<String, String> variables = Map.of(
+      "sdkCrate",
+      "aws_sdk_" + sdkId,
+      "operationName",
+      operationName,
+      "structureName",
+      structureName,
+      "snakeCaseOperationName",
+      snakeCaseOperationName,
+      "dafnyTypesModuleName",
+      dafnyTypesModuleName,
+      "variants",
+      toDafnyVariantsForStructure(outputShape).toString()
+    );
+
+    // Dafny maps smithy.api#Unit to ()
+    if (outputShape.getId() == ShapeId.from("smithy.api#Unit")) {
+      return TokenTree.of(
+        evalTemplate(
+          """
+          #[allow(dead_code)]
+          pub fn to_dafny(
+              value: &$sdkCrate:L::operation::$snakeCaseOperationName:L::$operationName:LOutput
```

Comment:
nit: If we're special casing unit conversion, we may as well avoid an unused variable warning.
```suggestion
              _value: &$sdkCrate:L::operation::$snakeCaseOperationName:L::$operationName:LOutput
```